src/invocation/ClusterService.ts

```diff
@@ -72,26 +72,27 @@ class ClusterService {
         , attemptPeriod: number
         , deferred: Q.Deferred<ClusterService>) {
         setImmediate(() => {
-            var currentAddress = this.addresses[index];
-            this.client.getConnectionManager().getOrConnect(currentAddress).then((connection: ClientConnection) => {
-                this.ownerConnection = connection;
-                this.initMemberShipListener().then(() => {
-                    deferred.resolve(this);
-                });
-            }).catch((e) => {
-                console.log(e);
-                if (index === this.addresses.length) {
-                    attemptLimit = attemptLimit - 1;
-                    if (attemptLimit === 0) {
-                        var error = new Error('Unable to connect to any of the following addresses ' + this.addresses);
-                        deferred.reject(error);
-                        return;
-                    } else {
-                        setTimeout(this.tryAddressIndex(0, attemptLimit, attemptPeriod, deferred), attemptPeriod);
-                    }
+            if (this.addresses.length <= index) {
```

Comment:
I did not notice that you commented on the commit itself, not the PR. Commit name is misleading. The fixing part is the second commit actually by adding feedback about success of connection.write. Let me squash the commits and fix the messages.